src/Microsoft.AspNetCore.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -306,8 +303,7 @@ private void BuildGlobbedLinkTags(TagHelperAttributeList attributes, TagHelperCo
                     continue;
                 }
 
-                attributes.SetAttribute(HrefAttributeName, url);
```

Comment:
Does the `<link>` tag helper also have a case where the `href` attribute may not be bound? If so, the previous `SetAttribute(HrefAttributeName, url)` would have done the Right Thing:tm: and add the attribute if it wasn't already present. But new code in `BuildLinkTag()` expects `attributes` will _always_ contain `href`.

Bottom line: May need an `addHref` and handling similar to that in the `<script>` tag helper. Unless of course the case is impossible here.